jmh-core/src/main/java/org/openjdk/jmh/generators/core/StateObjectHandler.java

```diff
@@ -104,7 +104,10 @@ public static void validateState(ClassInfo state) {
 
         boolean hasDefaultConstructor = false;
         for (MethodInfo constructor : state.getConstructors()) {
-            hasDefaultConstructor |= (constructor.getParameters().isEmpty() && constructor.isPublic());
+            if (constructor.getParameters().isEmpty()) {
```

Comment:
The changes here are not equivalent.  If any constructor`isEmpty `and `isPublic`, `hasDefaultConstructor` will always be `True`. 

But in the code change, the early exit could cause `hasDefaultConstructor` be `false` if there is a constructor `isEmpty` but not `isPublic` without iterating the default constructor yet.

Reply:
is it possible to have two default constructors with empty params? Maybe not in Java but in some alternative language?
As far I know this is prohibited.
If we can have only one empty param constructor then we can check only it for having a public modifier and then skip others

Reply:
I was thinking of the case the default constructor is declared explicitly with the `private` modifier. But you are right that a Java class could not have two default constructors with empty params, according to the rule of the method signature. The change is not equivalent, it still fulfills the same purpose.

Reply:
ok, so if we are fine here then I'll mark the comment as resolved